src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -59,6 +62,40 @@
                     totalCostInvoiceDetail.getTotalAmountBilledToDate().add(document.getInvoiceGeneralDetail().getCostShareAmount()));
         }
         
+        Award award = (Award) document.getInvoiceGeneralDetail().getAward();
```

Comment:
In the interest of clean clode and smaller functions, maybe these two blocks could broken into seperate functions?  Maybe putCustomAwardParameters() and putCustomInvoiceAccountDetais()?

Reply:
Adjusting.